### PR TITLE
Makefile: add target to generate bindings

### PR DESCRIPTION
**Pull request by rhatdan**
Add a `.generate-bindings` make target that only runs in the absence of
the `.generate-bindings` file or when a `types.go` file below
`pkg/bindings` has changed.

It does not execute anything yet is just a template to wire in the
generation in a future change.

Signed-off-by: Valentin Rothberg <rothberg@redhat.com>
Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

### CODE REVIEW

**Comment by rhatdan** [comment]
@baude @vrothberg @nalind PTAL

**Review by ashley-cui — COMMENTED** [review]
I think this creates a dependency on goimports  in $PATH now? If a user doesn't have goimports in their path, the imports will not be correctly done. Looks like the build tests are failing because of this.

 I also would prefer it if we kept the generated time in the generated files.

Also, generating the bindings touches go.mod and go.sum and throws a bunch of inconsistent vendoring errors at me. Not entirely sure what's going on, but that might be my fault.

The files are correctly generated though, so that's good

File: contrib/cirrus/setup_environment.sh [hunk 1]
```diff
@@ -177,6 +177,7 @@ case "$PODBIN_NAME" in
     *) die_unknown PODBIN_NAME
 esac
 
+make install.tools
```
Comment (cevich) [inline contrib/cirrus/setup_environment.sh:180]: Instead of this...try rebasing on https://github.com/containers/podman/pull/8967

The `install.tools` target is run during each of the "Build for XYZ" tasks and the entire $GOPATH is cached for that VM.  The change in my PR simply restores that cache for the `alt_build` task.  I did not do this previously out of fear of polluting something (call it paranoia).

File: pkg/bindings/generator/generator.go [hunk 2]
```diff
@@ -239,7 +236,7 @@ func main() {
 				closed = true
 
 				// go fmt file
-				gofmt := exec.Command("gofmt", "-w", "-s", out.Name())
+				gofmt := exec.Command("go", "fmt", out.Name())
```
Comment (baude) [inline pkg/bindings/generator/generator.go:239]: why did you remove this?

File: pkg/bindings/generate/types_systemd_options.go [hunk 3]
```diff
@@ -11,8 +11,6 @@ import (
 
 /*
 This file is generated automatically by go generate.  Do not edit.
-
-Created 2020-12-18 15:58:20.661450253 -0600 CST m=+0.000135779
```
Comment (baude) [inline pkg/bindings/generate/types_systemd_options.go:15]: why did you remove this?

File: pkg/bindings/network/types_create_options.go [hunk 4]
```diff
@@ -193,9 +191,9 @@ func (o *CreateOptions) WithIPRange(value net.IPNet) *CreateOptions {
 
 // GetIPRange
 func (o *CreateOptions) GetIPRange() net.IPNet {
-	var ipRange net.IPNet
+	var iPRange net.IPNet
```
Comment (baude) [inline pkg/bindings/network/types_create_options.go:194]: why?

File: pkg/bindings/containers/types_list_options.go [hunk 5]
```diff
@@ -40,7 +37,6 @@ func (o *ListOptions) ToParams() (url.Values, error) {
 		if !o.Changed(fieldName) {
 			continue
 		}
-		fieldName = strings.ToLower(fieldName)
```
Comment (baude) [inline pkg/bindings/containers/types_list_options.go:43]: this is likely your problem. looking into why.

File: pkg/bindings/network/types_create_options.go (hunk 4 above)
Comment (rhatdan) [reply pkg/bindings/network/types_create_options.go:194]: go generate does this not, me.

File: pkg/bindings/generate/types_systemd_options.go (hunk 3 above)
Comment (rhatdan) [reply pkg/bindings/generate/types_systemd_options.go:15]: Because this will cause us to have to update the bindings every time you run create bindings even if the bindings did not change.

File: pkg/bindings/generator/generator.go (hunk 2 above)
Comment (rhatdan) [reply pkg/bindings/generator/generator.go:239]: gofmt is not available in path on a lot of ci/cd images, go fmt seems to work, so I switched to it.

```
 go fmt [-n] [-x] [packages]

Fmt runs the command 'gofmt -l -w' on the packages named
by the import paths. It prints the names of the files that are modified.

For more about gofmt, see 'go doc cmd/gofmt'.
For more about specifying packages, see 'go help packages'.

The -n flag prints commands that would be executed.
The -x flag prints commands as they are executed.

The -mod flag's value sets which module download mode
to use: readonly or vendor. See 'go help modules' for more.

To run gofmt with specific options, run gofmt itself.

See also: go fix, go vet.
```

File: pkg/bindings/containers/types_list_options.go (hunk 5 above)
Comment (rhatdan) [reply pkg/bindings/containers/types_list_options.go:43]: Yup, this gets removed when you go generate, I tried to hard code this back in, but it caused other tools to break.
Strange that I can not get this same error to happen with podman-remote ps --last 1.

**Review by ashley-cui — COMMENTED** [review]
LGTM assuming tests pass!

**Review by vrothberg — COMMENTED** [review]
LGTM, thanks a lot @rhatdan !

**Comment by umohnani8** [comment]
LGTM
